property-scalaz/src/main/scala/testz/property.scala

```diff
@@ -45,34 +44,36 @@ object property {
     [F[_], I]
     (seed: (Array[Int], Int))
     (gen: Int => I)
-    (testGenerator: I => F[List[TestError]])
+    (testGenerator: I => F[TestResult])
     (numTestCases: Int)
-    (implicit F: Applicative[F]): F[List[TestError]] = {
+    (implicit F: Applicative[F]): F[TestResult] = {
     val rng = Well44497a.fromSeed(seed)
-    var acc: F[List[TestError]] = F.pure(Nil)
+    var acc: F[TestResult] = F.pure(Success)
     var i = 0
     while (i < numTestCases) {
-      acc = F.apply2(acc, testGenerator(gen(rng.nextInt)))(_ ++ _)
+      acc = F.apply2(acc, testGenerator(gen(rng.nextInt)))(TestResult.combine)
```

Comment:
Why not combine them as `List[TestResult]` (`traverse`)?
Why not short-circuit on failure?

Reply:
More error information is nice to have.

Reply:
Yes, but this way all runs will be in the same single list of errors. It is not important until you have more functionality (e.g. shrinking).

Reply:
I don't follow.